cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1623,6 +1623,10 @@ function Install-LoggingAgent {
     # installer binary and searched for flags to do this but found nothing. Oh
     # well.
     Log-Output ("Skip: Stackdriver logging agent is already installed")
+
+    # Restarting logging agent forces it to reconnect with the metadata 
+    # server. This makes the startup more reliable.
+    Restart-LoggingAgent
```

Comment:
I think it'd be better if stackdriverlogger and friends were set to automatically restart.

```bash
sc failure /?
DESCRIPTION:
        Changes the actions upon failure
USAGE:
        sc <server> failure [service name] <option1> <option2>...

OPTIONS:
        reset=   <Length of period of no failures (in seconds)
                  after which to reset the failure count to 0 (may be INFINITE)>
                  (Must be used in conjunction with actions= )
        reboot=  <Message broadcast before rebooting on failure>
        command= <Command line to be run on failure>
        actions= <Failure actions and their delay time (in milliseconds),
                  separated by / (forward slash) -- e.g., run/5000/reboot/800
                  Valid actions are <run|restart|reboot> >
                  (Must be used in conjunction with the reset= option)
```

Reply:
Good point - I filed a bug to track setting the logging service to auto-restart.

Reply:
Further look at the code the caller of this method actually invokes `Restart-LoggingAgent` afterwards.

```powershell
# Even if Stackdriver is already installed, the function will still [re]start the service.
  if (IsLoggingEnabled $kube_env) {
    Install-LoggingAgent
    Configure-LoggingAgent
    Restart-LoggingAgent
  }
```